flink-ml-lib/src/main/java/org/apache/flink/ml/feature/robustscaler/RobustScaler.java

```diff
@@ -0,0 +1,183 @@
+import org.apache.flink.ml.param.Param;
+import org.apache.flink.ml.util.ParamUtils;
+import org.apache.flink.ml.util.ReadWriteUtils;
+import org.apache.flink.streaming.api.datastream.DataStream;
+import org.apache.flink.table.api.Table;
+import org.apache.flink.table.api.bridge.java.StreamTableEnvironment;
+import org.apache.flink.table.api.internal.TableImpl;
+import org.apache.flink.types.Row;
+import org.apache.flink.util.Preconditions;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * Scale features using statistics that are robust to outliers.
+ *
+ * <p>This Scaler removes the median and scales the data according to the quantile range (defaults
+ * to IQR: Interquartile Range). The IQR is the range between the 1st quartile (25th quantile) and
+ * the 3rd quartile (75th quantile) but can be configured.
+ *
+ * <p>Centering and scaling happen independently on each feature by computing the relevant
+ * statistics on the samples in the training set. Median and quantile range are then stored to be
+ * used on later data using the transform method.
+ *
+ * <p>Standardization of a dataset is a common requirement for many machine learning estimators.
+ * Typically this is done by removing the mean and scaling to unit variance. However, outliers can
+ * often influence the sample mean / variance in a negative way. In such cases, the median and the
+ * interquartile range often give better results.
+ */
+public class RobustScaler
+        implements Estimator<RobustScaler, RobustScalerModel>, RobustScalerParams<RobustScaler> {
+    private final Map<Param<?>, Object> paramMap = new HashMap<>();
+
+    public RobustScaler() {
+        ParamUtils.initializeMapWithDefaultValues(paramMap, this);
+    }
+
+    @Override
+    public RobustScalerModel fit(Table... inputs) {
+        Preconditions.checkArgument(inputs.length == 1);
+        StreamTableEnvironment tEnv =
+                (StreamTableEnvironment) ((TableImpl) inputs[0]).getTableEnvironment();
+        DataStream<DenseVector> inputData =
+                tEnv.toDataStream(inputs[0])
+                        .map(
+                                (MapFunction<Row, DenseVector>)
+                                        value ->
+                                                ((Vector) value.getField(getInputCol())).toDense());
+        DataStream<RobustScalerModelData> modelData =
+                DataStreamUtils.aggregate(
+                        inputData,
+                        new QuantileAggregator(getRelativeError(), getLower(), getUpper()));
+        RobustScalerModel model =
+                new RobustScalerModel().setModelData(tEnv.fromDataStream(modelData));
+        ReadWriteUtils.updateExistingParams(model, getParamMap());
+        return model;
+    }
+
+    /**
+     * A stream operator to compute the medians and quantile ranges from feature column of the input
```

Comment:
It might be better to change "stream operator" to "function" or "aggregate function".